NeuroID/src/main/java/com/neuroid/tracker/NeuroID.kt

```diff
@@ -181,7 +184,14 @@ class NeuroID private constructor(
         }
     }
 
-    fun setUserID(userId: String): Boolean {
+    fun setUserID(userID: String): Boolean {
+        return setUserID(userID, true)
+    }
+
+    private fun setUserID(userId: String, checkForEmpty: Boolean): Boolean {
+        if (checkForEmpty && !validateUserId(userId)) {
+            return false
+        }
```

Comment:
If we don't check for empty, won't this skip validation? Meaning for startSession they could pass an invalid ID and we still allow it?

Reply:
startSession() calls setGenericUserID() that calls getOriginResult() which ultimately calls validateUserID(). validateUserID will ensure that a bad id is caught and in getOriginResult() will ensure that if a bad id is detected, it will be replaced with a random id.

Reply:
so are we really calling validate user twice?

Reply:
yeah I think so in case of properly formed ids for setUserID(). in case of badly formed ids, it will fail out once. for startSession() only once. I think I can move the validate to setUserID(String). let me take a look.